File: General/C++/Question_02a.cpp
//Find pairs in an integer array whose sum is equal to ten

/*This solution uses nested for loops which can be costly.
  Is there a better solution to this? */

#include<iostream>
#include<array>
#include<tuple>
#include<vector>

using namespace std;

vector<tuple <int, int> > tens(int *lst){

  vector<tuple <int, int> > answers;

  for(int x = 0; x < (sizeof(lst)-1); x++){
    for(int y = x+1; y < sizeof(lst); y++){
      if((lst[x] + lst[y]) == 10){
        answers.push_back(make_tuple(lst[x], lst[y]));
      }
    }
  }
  return answers;
}

//Test
int main(){
  
  int lst[12] = {2,3,4,7,8,9,1,10,12,18};
  vector<tuple <int, int> > test = tens(lst);
  
  for(auto i = test.begin(); i != test.end(); i++){
    cout << get<0>(*i) << "+" << get<1>(*i) << endl;
  }

  return 0;
}

File: General/Python/Question_02a.py
#Find pairs in an integer array whose sum is equal to 10.

#This solution uses nested for loops which can be costly.
#Is there a better solution to this?

def Tens(lst):
  answers = []
  for x in range(0, (len(lst)-1)):
    for y in range((x+1), len(lst)):
      if (lst[x] + lst[y] == 10):
        answers.append((lst[x],lst[y]))
  return answers

#Test
def main():
  test = Tens([1,2,6,8,9,4,10,0,11,-1])
  for i in range(0,len(test)):
    print(test[i])

if __name__ == '__main__':
  main()

          
